File: src/main/scala/net/degols/filesgate/libs/filesgate/storage/StorageMetadataApi.java
trait StorageMetadataApi {
  def update(fileMetadata: FileMetadata): Future[Try[UpdateOperation]]

  def save(fileMetadata: FileMetadata): Future[Try[SaveOperation]]

  def get(id: String): Future[Try[FileMetadata]]

  def delete(id: String): Future[Try[Boolean]]

  def listToRecover(priority: Int, quantity: Int): Future[Try[List[FileMetadata]]]
}

File: src/main/scala/net/degols/libs/filesgate/storage/StorageMetadataApi.scala
package net.degols.libs.filesgate.storage

import net.degols.libs.filesgate.orm.FileMetadata
import net.degols.libs.filesgate.pipeline.failurehandling.FailureHandlingMessage

import scala.concurrent.Future

trait StorageMetadataApi {
  def upsert(failureHandlingMessage: FailureHandlingMessage): Future[UpdateOperation]

  def upsert(fileMetadata: FileMetadata): Future[UpdateOperation]

  def get(id: String): Future[FileMetadata]

  def delete(id: String): Future[Boolean]
}
